### PR TITLE
Master libblockdev partitioning jobs

### PR DESCRIPTION
**Pull request by vojtechtrefny**

### CODE REVIEW

**Comment by vpodzime** [comment]
Jenkins, test this, please.

File: src/udiskslinuxpartition.c [hunk 1]
```diff
@@ -575,12 +648,19 @@ udisks_linux_partition_set_type_sync (UDisksLinuxPartition  *partition,
...
+      if (!bd_part_set_part_type (device_name, partition_name, type, &loc_error))
         {
-          g_prefix_error (error, "Error setting type for partition %s: ", partition_name);
+          g_set_error (error,
+                       UDISKS_ERROR,
+                       UDISKS_ERROR_FAILED,
+                       "Error setting partition type on %s: %s",
+                       udisks_block_get_device (block),
+                       loc_error->message);
+          udisks_simple_job_complete (UDISKS_SIMPLE_JOB (job), FALSE, (*error)->message);
```
Comment (vpodzime) [inline src/udiskslinuxpartition.c]: ``loc_error`` should be ``g_clear_error()``-ed here.

File: src/udiskslinuxpartition.c [hunk 2]
```diff
@@ -311,7 +326,13 @@ handle_set_flags (UDisksPartition       *partition,
...
                                    &error))
         {
-          g_dbus_method_invocation_take_error (invocation, error);
+          g_dbus_method_invocation_return_error (invocation,
+                                                 UDISKS_ERROR,
+                                                 UDISKS_ERROR_FAILED,
+                                                 "Error setting partition flags on %s: %s",
+                                                 udisks_block_get_device (block),
+                                                 error->message);
+          udisks_simple_job_complete (UDISKS_SIMPLE_JOB (job), FALSE, error->message);
```
Comment (vpodzime) [inline src/udiskslinuxpartition.c]: ``error`` needs to be ``g_clear_error()``-ed here or in the ``out:`` block. The same applies to almost all other modifications in this commit.

File: src/udiskslinuxpartitiontable.c [hunk 3]
```diff
@@ -32,6 +32,7 @@
 #include <glib/gstdio.h>
```
Comment (vpodzime) [inline src/udiskslinuxpartitiontable.c]: newly create**d** partitions

File: src/udiskslinuxpartitiontable.c [hunk 4]
```diff
@@ -428,7 +421,6 @@ udisks_linux_partition_table_handle_create_partition (UDisksPartitionTable   *ta
   g_free (wait_data);
   g_free (part_spec);
   g_free (error_message);
```
Comment (vpodzime) [inline src/udiskslinuxpartitiontable.c]: Is ``error_message`` still being used somewhere?

File: src/udiskslinuxpartitiontable.c [hunk 5]
```diff
@@ -391,32 +391,25 @@ udisks_linux_partition_table_handle_create_partition (UDisksPartitionTable   *ta
...
+          else
+            {
+              g_dbus_method_invocation_return_error (invocation,
+                                                     UDISKS_ERROR,
+                                                     UDISKS_ERROR_FAILED,
+                                                     "Error wiping newly created partition %s: %s",
+                                                     udisks_block_get_device (partition_block),
+                                                     error->message);
+              udisks_simple_job_complete (UDISKS_SIMPLE_JOB (job), FALSE, error->message);
+              g_clear_object (&partition_object);
```
Comment (vpodzime) [inline src/udiskslinuxpartitiontable.c]: ``error`` has to be ``g_clear_error()``-ed here too.

File: src/tests/dbus-tests/test_60_partitioning.py [hunk 6]
```diff
@@ -39,6 +39,13 @@ def test_create_mbr_partition(self):
```
Comment (vpodzime) [inline src/tests/dbus-tests/test_60_partitioning.py]: fail with an expected error...

Comment (vojtechtrefny) [inline src/tests/dbus-tests/test_60_partitioning.py]: Fixed

File: src/udiskslinuxpartition.c (hunk 1 above)
Comment (vojtechtrefny) [inline src/udiskslinuxpartition.c]: Already cleared at line 730

File: src/udiskslinuxpartitiontable.c (hunk 3 above)
Comment (vojtechtrefny) [inline src/udiskslinuxpartitiontable.c]: Fixed

File: src/udiskslinuxpartitiontable.c (hunk 4 above)
Comment (vojtechtrefny) [inline src/udiskslinuxpartitiontable.c]: Good catch, fixed.

File: src/udiskslinuxpartition.c (hunk 2 above)
Comment (vojtechtrefny) [inline src/udiskslinuxpartition.c]: Fixed.

**Comment by vpodzime** [comment]
Jenkins, test this, please.

**Pull request merged**